enhancements/authentication/oauth-proxy-share-configuration.md

```diff
@@ -0,0 +1,98 @@
+---
+title: oauth-proxy-shared-configuration
+authors:
+  - "@deads2k"
+reviewers:
+  - "@standa"
+approvers:
+  - "@sttts"
+creation-date: 2020-05-26
+last-updated: 2020-05-26
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - https://github.com/openshift/enhancements/pull/22  
+replaces:
+superseded-by:
+---
+
+# oauth-proxy Shared Configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+The [oauth-proxy](https://github.com/openshift/oauth-proxy) is a commonly used proxy which allows a naive application to 
+make use of in-cluster authentication and authorization to protect itself on an openshift platform without changing
+the application itself.
+This effectively makes the application appear as an extension to our platform, examples include: logging kibana, 
+monitoring grafana, etc.
+There are a few configuration values that are considered as related to platform configuration or security that the oauth-proxy
+could auto-detect.
+
+## Motivation
+
+If a customer wants to set a value on all oauth-proxy instances such as proxy configuration or `--cookie-expire` duration,
+there is no unified way to set that across all instances of the oauth-proxy.
+Having cluster-admins know every application using the oauth-proxy and figuring out how to configure them all seems 
+abusive.
+
+### Goals
+
+1. Make it easy to describe configuration for all oauth-proxy binaries.
+
+### Non-Goals
+
+
+## Proposal
+
+We will create an oauthproxies.config.openshift.io to ensure that read access can be widely granted without any risk of
+accidentally allowing reading of more private data (versus attaching to oauth.config.openshift.io for instance).
+Its .spec stanza contain typed fields for certain flags.
+
+The following flags for oauth-proxy are good targets for being consistent for all oauth-proxy instances in the cluster.
```

Comment:
So if I understand correctly, on our end the only thing we would need to change in cluster-monitoring-operator to just need to give the service accounts used by the proxy permissions to read oauthproxies.config.openshift.io.?

Reply:
> So if I understand correctly, on our end the only thing we would need to change in cluster-monitoring-operator to just need to give the service accounts used by the proxy permissions to read oauthproxies.config.openshift.io.?

If we did this, we would grant read access to all serviceaccounts. Will update the proposal.

Reply:
Just to clarify, this means that no additional permissions should be needed for our service accounts, right?